presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -447,6 +447,14 @@ public static double NaN()
         return Double.NaN;
     }
 
+    @Description("constant representing not-a-number in 32b float")
+    @ScalarFunction("fnan")
+    @SqlType(StandardTypes.FLOAT)
+    public static long floatNaN() // this should be removed once double -> float coersion is ready
```

Comment:
BTW now I think coercion won't solve the problem of lacking `fnan`. So maybe this function should not be removed even after implementing coercion. Or we will be stick using `CAST(nan() as FLOAT)`.